src/nevermined/Nfts.ts

```diff
@@ -512,4 +513,78 @@ export class Nfts extends Instantiable {
             throw Error(`Error listing ${ddo.shortId()} on secondary markets`)
         }
     }
+
+    public async lockBuyersPayments(
```

Comment:
You implemented three different methods that should create a waterfall flow one after the other @iamyxsh , and that method that is doing them step by step should be the one we have here. The `Nfts` module is a high-level interface used to call `composite methods` that run all these other methods in lower level modules  like `agreements.conditions.lockPayment`

Reply:
Th flow is the right one, but please implement one method that does these steps in this exact order, just altogether.

Reply:
Okay.